src/perception/vision_pipeline/vision_pipeline/trt_inference.py

```diff
@@ -0,0 +1,295 @@
+## The code below is adapted from 
+# https://github.com/wang-xinyu/tensorrtx/blob/master/yolov5/yolov5_trt.py
+
+# import python libraries
+import ctypes
+import time
+import cv2
+import numpy as np
+# import TensorRT libraries
+import pycuda.autoinit
+import pycuda.driver as cuda
+import tensorrt as trt
+
+class TensorWrapper(object):
+    """
+    description: A YOLOv5 class that wraps TensorRT ops, preprocess and postprocess ops.
+    """
+    def __init__(
+        self, 
+        engine_file_path: str, 
+        pluggin_file_path: str,
+        conf_thresh: float=0.5,
+        iou_thresh: float=0.4,
+    ):
+        # load in object pluggin file
+        ctypes.CDLL(pluggin_file_path)
+        # create a Context on this device,
+        self.ctx = cuda.Device(0).make_context()
+        stream = cuda.Stream()
+        TRT_LOGGER = trt.Logger(trt.Logger.INFO)
+        runtime = trt.Runtime(TRT_LOGGER)
+
+        # Deserialize the engine from file
+        f = open(engine_file_path, "rb")           
+        engine = runtime.deserialize_cuda_engine(f.read())
+        context = engine.create_execution_context()
```

Comment:
no `f.close`? might be better to use a context manager for this

Reply:
added f.close
not sure what a context manage would be for this, it just opens, saves a variable as what's inside the engine file i think

Reply:
https://www.geeksforgeeks.org/context-manager-in-python/